src/Wallet.ts

```diff
@@ -0,0 +1,111 @@
+import {fromSeed} from 'bip32'
+import {mnemonicToSeedSync} from 'bip39'
+import {decode, encodeForSigning} from 'ripple-binary-codec'
+import {deriveKeypair, generateSeed, verify} from 'ripple-keypairs'
+import ECDSA from './common/types/enums/ecdsa'
+import {SignedTransaction} from './common/types/objects'
+import {signOffline} from './transaction/sign'
+import {SignOptions} from './transaction/types'
+
+/**
+ * A utility for deriving a wallet composed of a keypair (publicKey/privateKey).
+ * A wallet can be derived from either a seed, mnemnoic, or entropy (array of random numbers).
+ * It provides functionality to sign/verify transactions offline.
+ */
+class Wallet {
+  readonly publicKey: string
+  readonly privateKey: string
+  private static readonly defaultAlgorithm: ECDSA = ECDSA.ed25519
+  private static readonly defaultDerivationPath: string = "m/44'/144'/0'/0/0"
+
+  constructor(publicKey: string, privateKey: string) {
+    this.publicKey = publicKey
+    this.privateKey = privateKey
+  }
+
+  /**
+   * Derives a wallet from a seed.
+   * @param {string} seed A string used to generate a keypair (publicKey/privateKey) to derive a wallet.
+   * @param {ECDSA} algorithm The digital signature algorithm to generate an address for.
+   * @returns {Wallet} A Wallet derived from a seed.
+   */
+  static fromSeed(seed: string, algorithm: ECDSA = Wallet.defaultAlgorithm): Wallet {
+    return Wallet.deriveWallet(seed, algorithm)
+  }
+
+  /**
+   * Derives a wallet from a mnemonic.
+   * @param {string} mnemonic A string consisting of words (whitespace delimited) used to derive a wallet.
+   * @param {string} derivationPath The path to derive a keypair (publicKey/privateKey) from a seed (that was converted from a mnemonic).
+   * @returns {Wallet} A Wallet derived from a mnemonic.
+   */
+  static fromMnemonic(
+    mnemonic: string,
+    derivationPath: string = Wallet.defaultDerivationPath
+  ): Wallet {
+    const seed = mnemonicToSeedSync(mnemonic)
+    const masterNode = fromSeed(seed)
+    const node = masterNode.derivePath(derivationPath)
+    if (node.privateKey === undefined) {
+      return undefined
+    }
```

Comment:
Don't return `undefined`. If this is an error, then throw an error, otherwise return `null`.